sql-plugin/src/main/spark320/scala/com/nvidia/spark/rapids/shims/HybridFileSourceScanExecMeta.scala

```diff
@@ -0,0 +1,216 @@
+  }
+}
+
+object HybridFileSourceScanExecMeta {
+  private val HYBRID_JAR_PLUGIN_CLASS_NAME = "com.nvidia.spark.rapids.hybrid.HybridPluginWrapper"
+
+  // Determines whether using HybridScan or GpuScan
+  def useHybridScan(conf: RapidsConf, fsse: FileSourceScanExec): Boolean = {
+    val isEnabled = if (conf.useHybridParquetReader) {
+      require(conf.loadHybridBackend,
+        "Hybrid backend was NOT loaded during the launch of spark-rapids plugin")
+      true
+    } else {
+      false
+    }
+    // For the time being, only support reading Parquet
+    lazy val isParquet = fsse.relation.fileFormat.getClass == classOf[ParquetFileFormat]
+    // Check if data types of all fields are supported by HybridParquetReader
+    lazy val allSupportedTypes = fsse.requiredSchema.exists { field =>
+      TrampolineUtil.dataTypeExistsRecursively(field.dataType, {
+        // For the time being, the native backend may return incorrect results over nestedMap
+        case MapType(kt, vt, _) if kt.isInstanceOf[MapType] || vt.isInstanceOf[MapType] => false
+        // For the time being, BinaryType is not supported yet
+        case _: BinaryType => false
+        case _ => true
+      })
+    }
+    // TODO: supports BucketedScan
+    lazy val noBucketedScan = !fsse.bucketedScan
+    // TODO: supports working along with Alluxio
+    lazy val noAlluxio = !AlluxioCfgUtils.enabledAlluxioReplacementAlgoConvertTime(conf)
+
+    isEnabled && isParquet && allSupportedTypes && noBucketedScan && noAlluxio
+  }
+
+  /**
+   * Check if runtimes are satisfied, including:
+   * - Spark distribution is not CDH or Databricks
+   * - Hybrid jar in the classpath
+   * - Java version is 1.8
+   * - Scala version is 2.12
+   * - Parquet V1 data source
+   */
+  def checkRuntimes(v1DataSourceList: String): Unit = {
+    checkNotRunningCDHorDatabricks()
+    checkHybridJarInClassPath()
+    checkJavaVersion()
+    checkScalaVersion()
+    checkV1Datasource(v1DataSourceList)
+  }
+
+  /**
+   * Check Spark distribution is not CDH or Databricks,
+   * report error if it is
+   */
+  private def checkNotRunningCDHorDatabricks(): Unit = {
+    if (VersionUtils.isCloudera || VersionUtils.isDataBricks) {
+      throw new RuntimeException("Hybrid feature does not support Cloudera/Databricks " +
+          "Spark releases, Please disable Hybrid feature by setting " +
+          "spark.rapids.sql.parquet.useHybridReader=false")
+    }
+  }
+
+  /**
+   * Check if the Hybrid jar is in the classpath,
+   * report error if not
+   */
+  private def checkHybridJarInClassPath(): Unit = {
+    try {
+      Class.forName(HYBRID_JAR_PLUGIN_CLASS_NAME)
+    } catch {
+      case e: ClassNotFoundException => throw new RuntimeException(
+        "Hybrid jar is not in the classpath, Please add Hybrid jar into the class path, or " +
+            "Please disable Hybrid feature by setting " +
+            "spark.rapids.sql.parquet.useHybridReader=false", e)
+    }
```

Comment:
I think this way to check the class only works on driver side.
Do we need to check on executor side as well?

Reply:
> I think this way to check the class only works on driver side.

Yes.

> Do we need to check on executor side as well?

Yes.  Will check.

Reply:
Done.